central/reports/scheduler/v2/reportgenerator/report_gen_impl.go

```diff
@@ -158,7 +160,7 @@ func (rg *reportGeneratorImpl) generateReportAndNotify(req *ReportRequest) error
 
 func (rg *reportGeneratorImpl) saveReportData(configID, reportID string, data *bytes.Buffer) error {
 	if data == nil {
-		data = bytes.NewBuffer([]byte{})
+		return errors.Errorf("Unexpected empty downloadable report for report config=%q, id=%q", configID, reportID)
```

Comment:
Data can be nil in expected ways e.g. due to collection that does not match or access scope.
I would recommend changing the wording a bit like `No data found for report config...`

Reply:
It should not be nil here when saving data to blob.